bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -174,4 +175,24 @@
      */
     void handleRemoval();
 
+    /**
+     * This method should return a unique identifier of the device handled by this {@link ThingHandler}.
```

Comment:
Did you check the example implementation of `getDeviceId()`in the HueLightHandler or HueBridgeHandler for example?

For the light: 
```` 
return light.getUniqueID();
```` 
or for the bridge: 
```` 
return bridge.getFullConfig().getConfig().getMACAddress().replaceAll(":", "").toLowerCase();
````  
The value is not taken out of a property of the thing, because these properties could be set manually during creation of that thing. I was also thinking to move this method from the handler to the thing itself, but I refrained. It almost alway requires physical interaction with the device to get some unique identifier.